src/data/matrix/basic.lean

```diff
@@ -737,6 +737,22 @@ end
 
 end ring
 
+section comm_ring
+
+variables [comm_ring α]
+
+lemma mul_vec_smul_assoc (A : matrix m n α) (b : n → α) (a : α) :
+  A.mul_vec (a • b) = a • (A.mul_vec b) :=
+begin
+  ext i,
+  change dot_product (A i) (a • b) = _,
+  rw [← smul_mul_vec_assoc A b a, dot_product_comm],
+  change _ = dot_product ((a • A) i) b,
+  simp only [dot_product_comm, dot_product_smul, pi.smul_apply],
+end
```

Comment:
Here's a shorter proof, in the style of `mul_vec_neg` above :)
```suggestion
by { ext, apply dot_product_smul }
```

Reply:
Oh I see, thanks for the suggestion. Is this lemma still needed then?

Reply:
Yes, definitely - the cost of making `mul_vec` an semi-opaque wrapper for `dot_product` is that you have to make copies of all the lemmas. The fact that two proofs are definitionally the same is not an indication that one of them is redundant.